pysingfel/particle.py

```diff
@@ -27,6 +29,12 @@ def __init__(self, *fname):
         self.num_atom_types = None  # number of atom types
         self.ff_table = None  # form factor table -> atom_type x qSample
         
+        # Masks and solvent
+        self.hydration_layer_thickness = 4.0 / 10**10    # in meter
+        self.mesh_voxel_size           = 2.0 / 10**10    # in meter
+        self.mesh = None         # real space mesh for mask definitions
```

Comment:
Does the mesh have any correspondence with the mesh used for the diffraction data?
Do they need to have the same number of voxels?

Reply:
at this point they are independent, this is the most naive and brute-force implementation. Tonight or tomorrow I'll try to implement a downsampling step so the grid corresponds to the Fourier sampling. Pretty much what you suggested the other day actually: FFT+filter+IFFT